### DISCUSSION TITLE
MVVM / Data Binding / ObservableCollection - Inconsistent behaviours in different implementations, as well as insufficient documentation

### DISCUSSION THREAD

**Discussion by AmSmart** [xamarin/Xamarin.Forms#13048] [Q&A]
From my experience with Xamarin.Forms, the data binding model has been very temperamental based on different scenarios. In some cases, my ObservableCollection only updates on the UI when I re-assign to a new object. In other cases, it updates the UI until I assign it to a new object reference. I also believe that it behaves as expected in both scenarios at some other times, I know that there is a difference between listening for  INotifyPropertyChanged and INotifyCollectionChanged but I have not found any documentation to explain to detail what could be the cause of these Inconsistent behaviours.

Can anyone help, please?

**Comment by GalaxiaGuy** [comment]
Do you have a specific example that does not work? The most likely explanation is a specific property that just isn't paying attention to `INotifyCollectionChanged` when perhaps it should be.
